src/adapters/philips_hue/lights.rs

```diff
@@ -0,0 +1,146 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use foxbox_taxonomy::api::Error;
+use foxbox_taxonomy::manager::*;
+use foxbox_taxonomy::services::*;
+use foxbox_taxonomy::values::{ Type };
+use super::*;
+use super::hub_api::HubApi;
+use std::collections::HashSet;
+use std::sync::Arc;
+use traits::Controller;
+
+const CUSTOM_PROPERTY_MANUFACTURER: &'static str = "manufacturer";
+const CUSTOM_PROPERTY_MODEL: &'static str = "model";
+const CUSTOM_PROPERTY_TYPE: &'static str = "type";
+
+#[derive(Clone)]
+pub struct Light {
+    api: Arc<HubApi>,
+    hub_id: String,
+    light_id: String,
+    service_id: Id<ServiceId>,
+    pub get_available_id: Id<Getter>,
+    pub get_power_id: Id<Getter>,
+    pub set_power_id: Id<Setter>,
+}
+
+impl Light {
+    pub fn new(api: Arc<HubApi>, hub_id: &str, light_id: &str)
+        -> Self
+    {
+        Light {
+            api: api,
+            hub_id: hub_id.to_owned(),
+            light_id: light_id.to_owned(),
+            service_id: create_light_id(&hub_id, &light_id),
+            get_available_id: create_getter_id("available", &hub_id, &light_id),
```

Comment:
We came across this topic in the past: This ```availability``` extension might evolve into something that every service exposes, because it signifies the distinction between "Foxbox knows about this device, but it's currently unreachable" and "Foxbox can actually talk to this device".